core/src/banking_stage/transaction_scheduler/transaction_state_container.rs

```diff
@@ -133,8 +133,21 @@ impl<Tx: TransactionWithMeta> StateContainer<Tx> for TransactionStateContainer<T
             .map(|state| state.transaction_ttl())
     }
 
-    fn push_id_into_queue(&mut self, priority_id: TransactionPriorityId) -> bool {
-        self.push_id_into_queue_with_remaining_capacity(priority_id, self.remaining_capacity())
+    fn push_ids_into_queue(
+        &mut self,
+        priority_ids: impl Iterator<Item = TransactionPriorityId>,
+    ) -> usize {
+        for id in priority_ids {
+            self.priority_queue.push(id);
+        }
+        let num_dropped = self.priority_queue.len().saturating_sub(self.capacity);
```

Comment:
I think we also want to enforce that the slab (`id_to_transaction_state`) length is also not more than `self.capacity` after pushing ids into the queue right? If we use `priority_queue` length, we might actually already have some transactions that are in the slab but aren't currently in the priority queue because they're scheduled. So this could mean that after pushing 64 txs into the slab and all checks pass, we could have at least 64 transactions scheduled and so there is space for all of those transactions in the priority queue. So `ReceiveAndBuffer` would run another pass and could fill an extra 64 txs into the slab again.

Reply:
Yeah you're right about the scheduled txs not in queue - we should be using the slab's len here because it should always be >= to priority_queue len. i.e. we never have something in queue that is not in slab.

Reply:
[788f144](https://github.com/anza-xyz/agave/pull/4506/commits/788f144e5b7435d8b8340801f80c57cda8c7c6d5)

Added test to test this as well, with all txs in queue popped but not removed from map.
Realistically that will not happen due to our CU throttling, we don't have **everything** scheduled.